scaleway/resource_container.go

```diff
@@ -0,0 +1,401 @@
+				}, false),
+			},
+			"registry_image": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The scw registry image address",
+			},
+			"max_concurrency": {
+				Type:             schema.TypeInt,
+				Optional:         true,
+				Computed:         true,
+				Description:      "The maximum the number of simultaneous requests your container can handle at the same time. Defaults to 50.",
+				ValidateDiagFunc: validateMaxLimit(containerMaxLimit),
+			},
+			"domain_name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The container domain name.",
+			},
+			"protocol": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The communication protocol. Defaults to http1.",
+				Default:     container.ContainerProtocolHTTP1.String(),
+				ValidateFunc: validation.StringInSlice([]string{
+					container.ContainerProtocolH2c.String(),
+					container.ContainerProtocolHTTP1.String(),
+					container.ContainerProtocolUnknownProtocol.String()},
+					false),
+			},
+			"port": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Optional:    true,
+				Description: "The port to expose the container. Defaults to 8080",
+			},
+
+			"redeploy": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Allow deploy container",
+				Default:     false,
+			},
+			"region": regionSchema(),
+			// computed
+			"status": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The container status",
+				Computed:    true,
+			},
+			"cron_status": {
+				Type:        schema.TypeString,
+				Description: "The cron status",
+				Computed:    true,
+			},
+			"error_message": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The error description",
+			},
+		},
+	}
+}
+
+func resourceScalewayContainerCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api, region, err := containerAPIWithRegion(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	namespaceID := d.Get("namespace_id")
+	// verify name space state
+	_, err = api.WaitForNamespace(&container.WaitForNamespaceRequest{
+		NamespaceID:   expandID(namespaceID),
+		Region:        region,
+		Timeout:       scw.TimeDurationPtr(defaultRegistryNamespaceTimeout),
+		RetryInterval: DefaultWaitRetryInterval,
+	}, scw.WithContext(ctx))
+	if err != nil {
+		return diag.Errorf("unexpected namespace error: %s", err)
+	}
+
+	req, err := setCreateContainerRequest(d, region)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	res, err := api.CreateContainer(req, scw.WithContext(ctx))
+	if err != nil {
+		return diag.Errorf("creation container error: %s", err)
+	}
+
+	d.SetId(newRegionalIDString(region, res.ID))
+
+	// verify name space state
+	_, err = api.WaitForNamespace(&container.WaitForNamespaceRequest{
```

Comment:
Should we wait for the Container AND the Namespace ?

Reply:
This is the normal check when for resource dependency. I'm gonna try the test if it is really necessary.